.github/workflows/analyze-maps-from-circleci.yml

```diff
@@ -0,0 +1,122 @@
+name: Analyze Maps from CircleCI
+
+on:
+  pull_request:
+
+# AWS OIDC Related Permissions
+permissions:
+  id-token: write # This is required for requesting the JWT
+  contents: read  # This is required for actions/checkout
+  # AppMap Specific Permissions
+  actions: read
+  checks: write
+  pull-requests: write
+
+jobs:
+  download_and_archive_base_revision:
+    timeout-minutes: 30
+    env:
+      appmap_storage_bucket: circleci-appmaps
+      base_sha: ${{ github.event.pull_request.base.sha }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install AppMap tools
+        uses: getappmap/install-action@v1
+        with:
+          install-appmap-library: false
+
+      - name: Check for existing baseline archive
+        id: check_for_baseline
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        continue-on-error: true  
+        run: |
+          appmap restore --revision ${base_sha} --github-repo ${GITHUB_REPOSITORY} --exact 
```

Comment:
While this is fine and works - if we had a `--check` flag or something to check the existence of the archive vs doing the full download that would avoid some unnecessary network.

Reply:
Cool - i'll take a look!  Thanks for the link